esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2334,14 +2328,10 @@ void CFileSprayEx::getDropZoneInfoByDestPlane(double clientVersion, const char*
         Owned<IPropertyTree> match = getHostGroup(hostGroup,true);
         if (!hostip.isEmpty())
         {
-            Owned<IPropertyTreeIterator> hostIter = match->getElements("hosts");
-            ForEach (*hostIter)
-            {
-                const char *knownIP = hostIter->query().queryProp(nullptr);
-                if (strcmp(knownIP, hostip.str())==0)
-                    return;
-            }
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "destip %s is not valid storage plane %s", hostip.str(), destPlane);
+            VStringBuffer xexpr(".[hosts='%s']", hostip.str());
+            if (!match->hasProp(xexpr))
+                throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "destip %s is not valid storage plane %s", hostip.str(), destPlane);
+            return;
```

Comment:
probably better not to include the return - in case someone added something to the end of the function.